lib/schema/map.js

```diff
@@ -8,15 +8,21 @@ const MongooseMap = require('../types/map');
 const SchemaMapOptions = require('../options/schemaMapOptions');
 const SchemaType = require('../schemaType');
 const createJSONSchemaTypeDefinition = require('../helpers/createJSONSchemaTypeDefinition');
+const MongooseError = require('../error/mongooseError');
+const Schema = require('../schema');
+const utils = require('../utils');
 
 /*!
  * ignore
  */
```

Comment:
`SchemaMap` is public right? This should likely also have a proper jsdoc then like the other schema types (though not directly related to this PR)